### ISSUE TITLE
memory validation : your system has only -1MB memory.

### ISSUE THREAD

**Issue by medyagh** [kubernetes/minikube#8958]
while doing this PR 
https://github.com/kubernetes/minikube/pull/8953

I saw this
```
2020-08-10T20:13:28.2056765Z === RUN   TestFunctional/serial/SoftStart
2020-08-10T20:13:28.2065216Z     functional_test.go:252: (dbg) Run:  ./minikube-windows-amd64.exe start -p functional-20200810200921-8436
2020-08-10T20:13:34.9476059Z     functional_test.go:252: (dbg) Non-zero exit: ./minikube-windows-amd64.exe start -p functional-20200810200921-8436: exit status 78 (6.7410182s)
2020-08-10T20:13:34.9478123Z         
2020-08-10T20:13:34.9480963Z         -- stdout --
2020-08-10T20:13:34.9482954Z         	* [functional-20200810200921-8436] minikube v1.12.2 on Microsoft Windows 10 Enterprise N 10.0.18363 Build 18363
2020-08-10T20:13:34.9484745Z         	  - KUBECONFIG=C:\Users\jenkins\actions-runner\_work\minikube\minikube\minikube_binaries\testhome\kubeconfig
2020-08-10T20:13:34.9486792Z         	  - MINIKUBE_HOME=C:\Users\jenkins\actions-runner\_work\minikube\minikube\minikube_binaries\testhome
2020-08-10T20:13:34.9488589Z         	* Using the docker driver based on existing profile
2020-08-10T20:13:34.9490450Z         	* To suppress memory validations you can use --force flag.
2020-08-10T20:13:34.9492518Z         
2020-08-10T20:13:34.9494316Z         -- /stdout --
2020-08-10T20:13:34.9498039Z         ** stderr ** 
2020-08-10T20:13:34.9498444Z         	! Failed to verify system memory limits.
2020-08-10T20:13:34.9503086Z         	! Your system has only -1MB memory. This might not work minimum required is 2000MB.
2020-08-10T20:13:34.9506817Z         	! Increase Docker for Desktop memory to at least 2.5GB or more:
2020-08-10T20:13:34.9510248Z         				
2020-08-10T20:13:34.9513581Z         		Docker for Desktop > Settings > Resources > Memory
2020-08-10T20:13:34.9515457Z         	
2020-08-10T20:13:34.9517904Z         	
2020-08-10T20:13:34.9519710Z         	! Failed to verify system memory limits.
2020-08-10T20:13:34.9521631Z         	! Your system has only -1MB memory. This might not work minimum required is 2000MB.
2020-08-10T20:13:34.9523749Z         	! Increase Docker for Desktop memory to at least 2.5GB or more:
2020-08-10T20:13:34.9525792Z         				
2020-08-10T20:13:34.9528137Z         		Docker for Desktop > Settings > Resources > Memory
2020-08-10T20:13:34.9530091Z         	
2020-08-10T20:13:34.9531866Z         	
2020-08-10T20:13:34.9534437Z         	X Requested memory allocation 4000MB is more than your system limit -1MB. Try specifying a lower memory:
2020-08-10T20:13:34.9536258Z         	
2020-08-10T20:13:34.9539078Z         		miniube start --memory=-0.5mb				
2020-08-10T20:13:34.9541407Z         	
2020-08-10T20:13:34.9543190Z         	
2020-08-10T20:13:34.9544981Z         
2020-08-10T20:13:34.9547150Z         ** /stderr **
```

**Comment by medyagh** [comment]
the docker system info  was showing this number

"17299783680"

**Comment by medyagh** [comment]
Also might be related

```

-08-10T23:02:41.8218226Z === CONT  TestFunctional/parallel/DryRun
2020-08-10T23:02:41.8220716Z     functional_test.go:430: (dbg) Non-zero exit: ./minikube-windows-amd64.exe start -p functional-20200810225634-11176 --dry-run --memory 250MB --alsologtostderr --driver=docker: exit status 78 (4.0925143s)
2020-08-10T23:02:41.8224818Z         
2020-08-10T23:02:41.8227789Z         -- stdout --
2020-08-10T23:02:41.8232727Z         	* [functional-20200810225634-11176] minikube v1.12.2 on Microsoft Windows 10 Enterprise N 10.0.18363 Build 18363
2020-08-10T23:02:41.8235596Z         	  - KUBECONFIG=C:\Users\jenkins\actions-runner\_work\minikube\minikube\minikube_binaries\testhome\kubeconfig
2020-08-10T23:02:41.8238325Z         	  - MINIKUBE_HOME=C:\Users\jenkins\actions-runner\_work\minikube\minikube\minikube_binaries\testhome
2020-08-10T23:02:41.8241637Z         	* Using the docker driver based on existing profile
2020-08-10T23:02:41.8244818Z         
2020-08-10T23:02:41.8248802Z         -- /stdout --
2020-08-10T23:02:41.8250093Z         ** stderr ** 
2020-08-10T23:02:41.8254774Z         	I0810 23:02:38.087741    7316 out.go:191] Setting JSON to false
2020-08-10T23:02:41.8261085Z         	I0810 23:02:38.090752    7316 start.go:100] hostinfo: {"hostname":"mini-test4","uptime":2249184,"bootTime":1594851374,"procs":186,"os":"windows","platform":"Microsoft Windows 10 Enterprise N","platformFamily":"Standalone Workstation","platformVersion":"10.0.18363 Build 18363","kernelVersion":"","virtualizationSystem":"","virtualizationRole":"","hostid":"0a32a819-8625-428e-ba51-d3c302da5c9a"}
2020-08-10T23:02:41.8264909Z         	W0810 23:02:38.090752    7316 start.go:108] gopshost.Virtualization returned error: not implemented yet
2020-08-10T23:02:41.8269034Z         	I0810 23:02:38.118753    7316 driver.go:287] Setting default libvirt URI to qemu:///system
2020-08-10T23:02:41.8274711Z         	I0810 23:02:38.350752    7316 docker.go:87] docker version: linux-19.03.8
2020-08-10T23:02:41.8277152Z         	I0810 23:02:38.364773    7316 start.go:229] selected driver: docker
2020-08-10T23:02:41.8281767Z         	I0810 23:02:38.364773    7316 start.go:635] validating driver "docker" against &{Name:functional-20200810225634-11176 KeepContext:false EmbedCerts:false MinikubeISO: KicBaseImage:gcr.io/k8s-minikube/kicbase:v0.0.11@sha256:6fee59db7d67ed8ae6835e4bcb02f32056dc95f11cb369c51e352b62dd198aa0 Memory:4000 CPUs:2 DiskSize:20000 VMDriver: Driver:docker HyperkitVpnKitSock: HyperkitVSockPorts:[] DockerEnv:[] InsecureRegistry:[] RegistryMirror:[] HostOnlyCIDR:192.168.99.1/24 HypervVirtualSwitch: HypervUseExternalSwitch:false HypervExternalAdapter: KVMNetwork:default KVMQemuURI:qemu:///system KVMGPU:false KVMHidden:false DockerOpt:[] DisableDriverMounts:false NFSShare:[] NFSSharesRoot:/nfsshares UUID: NoVTXCheck:false DNSProxy:false HostDNSResolver:true HostOnlyNicType:virtio NatNicType:virtio KubernetesConfig:{KubernetesVersion:v1.18.3 ClusterName:functional-20200810225634-11176 APIServerName:minikubeCA APIServerNames:[] APIServerIPs:[] DNSDomain:cluster.local ContainerRuntime:docker CRISocket: NetworkPlugin: FeatureGates: ServiceCIDR:10.96.0.0/12 ImageRepository: LoadBalancerStartIP: LoadBalancerEndIP: ExtraOptions:[] ShouldLoadCachedImages:true EnableDefaultCNI:false CNI: NodeIP: NodePort:8441 NodeName:} Nodes:[{Name: IP:172.17.0.2 Port:8441 KubernetesVersion:v1.18.3 ControlPlane:true Worker:true}] Addons:map[ambassador:false dashboard:false default-storageclass:true efk:false freshpod:false gcp-auth:false gvisor:false helm-tiller:false ingress:false ingress-dns:false istio:false istio-provisioner:false kubevirt:false logviewer:false metallb:false metrics-server:false nvidia-driver-installer:false nvidia-gpu-device-plugin:false olm:false pod-security-policy:false registry:false registry-aliases:false registry-creds:false storage-provisioner:true storage-provisioner-gluster:false] VerifyComponents:map[apiserver:true apps_running:true default_sa:true system_pods:true]}
2020-08-10T23:02:41.8284631Z         	I0810 23:02:38.364773    7316 start.go:646] status for docker: {Installed:true Healthy:true NeedsImprovement:false Error:<nil> Fix: Doc:}
2020-08-10T23:02:41.8292744Z         	I0810 23:02:38.376749    7316 cli_runner.go:109] Run: docker system info --format "{{json .}}"
2020-08-10T23:02:41.8293215Z         	I0810 23:02:39.870675    7316 cli_runner.go:109] Run: docker system info --format "{{json .}}"
2020-08-10T23:02:41.8296382Z         	I0810 23:02:41.307118    7316 cli_runner.go:109] Run: docker system info --format "{{json .}}"
2020-08-10T23:02:41.8299911Z         	X Requested memory allocation 250MB is less than the usable minimum of <no value>MB
```

**Comment by medyagh** [comment]
this happend on hyperv too

**Comment by medyagh** [comment]
```
                                                                                                                                                                                    PS C:\Users\jenkins\actions-runner\_work\minikube\minikube\minikube_binaries> Measure-Command {.\minikube-windows-amd64 start --driver=hyperv --memory=4000MB}                      ! Failed to verify system memory limits.                                                                                                                                            ! Your system has only -1MB memory. This might not work minimum required is 2000MB.                                                                                                 X Requested memory allocation 4000MB is more than your system limit -1MB. Try specifying a lower memory:
        miniube start --memory=-0.5mb




Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 6
Milliseconds      : 624
Ticks             : 66240408
TotalDays         : 7.66671388888889E-05
TotalHours        : 0.00184001133333333
TotalMinutes      : 0.11040068
TotalSeconds      : 6.6240408
```

**Comment by afbjorklund** [comment]
memoryLimits returns -1 when there is an error. you can see the error in the logs.

```go
        s, c, err := memoryLimits(drvName)
        if err != nil {
                glog.Warningf("Unable to query memory limits: %v", err)
                out.WarningT("Failed to verify system memory limits.")
        }
        if s < 2200 {
                out.WarningT("Your system has only {{.memory_amount}}MB memory. This might not work minimum required is 2000MB.", out.V{"memory_amount": s})
        }
```

It needs to stop handling after the "err != nil", or look for s == -1

**Comment by tstromberg** [comment]
Was this fixed?

**Comment by ilya-zuyev** [comment]
Was fixed by https://github.com/kubernetes/minikube/pull/8959

**Comment by priyawadhwa** [comment]
Will close, as this was fixed by #8959
